src/Fls.Results/OperationResult.cs

```diff
@@ -171,5 +173,51 @@ public static async Task<IOperationResult<T>> BindErrorAsync<T>(this Task<IOpera
         {
             return await (await source).BindErrorAsync(bind, getMessage, exceptionCode);
         }
+
+        public static IOperationResult<T> ToResult<T>(this T source)
+        {
+            return Success(source);
+        }
+        
+        public static IOperationResult<T> IfError<T>(this IOperationResult<T> source, Func<IOperationResult<T>> returnResult)
+        {
+            return source.Match(
+                _ => source,
+                (_, error) => returnResult(),
+                failure => returnResult()
+            );
+        }
+
+        public static async Task<IOperationResult<T>> IfErrorAsync<T>(this Task<IOperationResult<T>> source, Func<IOperationResult<T>> returnResult)
+        {
+            return (await source).IfError(returnResult);
+        }
+
+        public static IOperationResult<T[]> All<T>(this IEnumerable<Func<IOperationResult<T>>> source)
+        {
+            return source.Aggregate(new List<T>().ToResult(), (a, f) => a.Bind(aList => f().Bind(newRes =>
+            {
+                aList.Add(newRes);
+                return aList.ToResult();
+            }))).Bind(x => x.ToArray().ToResult());
+        }
+
+        public static IOperationResult<T[]> Any<T>(this IEnumerable<Func<IOperationResult<T>>> source, T returnAsError)
+        {
+            return source.Aggregate(new List<T>().ToResult(), (a, f) => a.Bind(aList => f().IfError(() => Success(returnAsError)).Bind(newRes =>
+            {
+                    aList.Add(newRes);
+                    return aList.ToResult();
+            }))).Bind(x => x.ToArray().ToResult());
+        }
+
+        public static IOperationResult<T[]> Any<T>(this IEnumerable<Func<IOperationResult<T>>> source)
+        {
+            return source.Aggregate(new List<T>().ToResult(), (a, f) => a.Bind(aList => f().Bind(newRes =>
```

Comment:
This method should really return `T` and it's an opposite of `All`. First success matched success is returned ifError the rest.